tests/acceptance/pageObjects/FilesPageElement/fileActionsMenu.js

```diff
@@ -0,0 +1,223 @@
+      const actionsDropdownSelector = this.elements.itemActionsDropdown.selector
+      const actionSelector = this.elements[action + 'ButtonInFileRow'].selector
+
+      return `${actionsDropdownSelector}${actionSelector}`
+    },
+    /**
+     * perform one of the main file actions
+     *
+     * @param {string} action delete|share|rename|download
+
+     * @throws Error
+     * @returns {*}
+     */
+    performFileAction: function (action) {
+      const fileActionBtnSelectorXpath = this.getActionSelector(action)
+      return this
+        .useXpath()
+        .waitForElementVisible(fileActionBtnSelectorXpath)
+        .click(fileActionBtnSelectorXpath)
+        .useCss()
+    },
+    /**
+     * returns the disabled state of given action
+     *
+     * @param {string} action
+     * @returns {Promise<boolean>}
+     */
+    getActionDisabledAttr: async function (action) {
+      let disabledState
+      const btnSelector = this.getActionSelector(action)
+      await this
+        .api.element('xpath', btnSelector, result => {
+          // action is disabled when not visible in dropdown menu
+          disabledState = result.status === -1
+        })
+
+      return disabledState
+    },
+    /**
+     * deletes resource using fileActions 'delete' button
+     * @returns {Promise<*>}
+     */
+    delete: async function () {
+      this.performFileAction(this.FileAction.delete)
+      await this.api.page.FilesPageElement.filesList().confirmDeletion()
+      return this
+    },
+    /**
+     * @param {string} toName
+     * @param {boolean} expectToSucceed
+     * @return {*}
+     */
+    rename: async function (toName, expectToSucceed = true) {
+      await this.initAjaxCounters()
+        .useXpath()
+        .performFileAction(this.FileAction.rename)
+        .waitForElementVisible('@renameFileConfirmationBtn')
+        .waitForAnimationToFinish()
+        .clearValue('@renameFileInputField')
+        .setValue('@renameFileInputField', toName)
+        .click('@renameFileConfirmationBtn')
+        .waitForOutstandingAjaxCalls()
+        .useCss()
+
+      if (expectToSucceed) {
+        await this.waitForElementNotVisible('@renameFileConfirmationDialog')
+      }
+
+      return this
+    },
+    /**
+     * opens sharing dialog for given resource
+     * assumes filesAction menu for the resource to be opened
+     * @return {*}
+     */
+    openCollaboratorsDialog: function () {
+      const api = this.api.page.FilesPageElement
+      api.appSideBar().closeSidebar(500)
+      this
+        .useXpath()
+        .performFileAction(this.FileAction.share)
+        .waitForElementVisible('@sharingSideBar')
+        .useCss()
+      return api.sharingDialog()
+    },
+    /**
+     * opens links dialog for given resource
+     * assumes fileActionsMenu to be opened, as the its moves to links tab from file-actions share option
+     *
+     * @return {*}
+     */
+    openLinksDialog: function () {
```

Comment:
that is not part of the file-action menu

Reply:
moved to filesList